src/main/kotlin/three/consulting/epoc/controller/CustomerController.kt

```diff
@@ -1,16 +1,18 @@
 package three.consulting.epoc.controller
 
 import org.springframework.http.MediaType.ALL_VALUE
-import org.springframework.web.bind.annotation.GetMapping
-import org.springframework.web.bind.annotation.PathVariable
-import org.springframework.web.bind.annotation.RequestMapping
-import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.bind.annotation.*
+import three.consulting.epoc.dto.CustomerDTO
 import three.consulting.epoc.service.CustomerService
+import javax.validation.Valid
 
 @RestController
 @RequestMapping(path = ["/customer"])
 class CustomerController(private val customerService: CustomerService) {
 
     @GetMapping(value = ["/{customerId}"], consumes = [ALL_VALUE])
     fun getCustomerForId(@PathVariable customerId: Long) = customerService.findCustomerForId(customerId)
+
+    @PostMapping()
+    fun createCustomer(@Valid @RequestBody customer: CustomerDTO) = customerService.createCustomer(customer)
```

Comment:
Would it be better to also change the `customer` to `customerPayload`or just `payload` for more informative naming? @LeadingMoominExpert

Reply:
You can if you want. In this layer only the dto classes are used so there's no conflicts, but it's up to you